integration/pkcs11/pkcs11_test.go

```diff
@@ -0,0 +1,411 @@
+	adminPubKey, adminCSR, adminSerial := createCSR(ctx, sess, orgName, "admin")
+
+	// Retrieves org CA cert
+	orgCAPath := network.PeerOrgCADir(network.Organization(orgName))
+	caBytes, err := ioutil.ReadFile(filepath.Join(orgCAPath, "ca.org1.example.com-cert.pem"))
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Updating the peer signcerts")
+
+	newOrdererPemCert := buildCert(network, caBytes, orgCAPath, peerCSR, peerSerial, peerPubKey)
+
+	updateMSPFolder(network.PeerLocalMSPDir(peer), "peer.org1.example.com-cert.pem", newOrdererPemCert)
+
+	By("Updating the peer admin user signcerts")
+	newAdminPemCert := buildCert(network, caBytes, orgCAPath, adminCSR, adminSerial, adminPubKey)
+
+	orgAdminMSPPath := network.PeerUserMSPDir(peer, "Admin")
+	updateMSPFolder(orgAdminMSPPath, "Admin@org1.example.com-cert.pem", newAdminPemCert)
+
+}
+
+func configureOrdererPKCS11(ctx *pkcs11.Ctx, sess pkcs11.SessionHandle, network *nwo.Network) {
+
+	orderer := network.Orderer("orderer")
+	orgName := "orderer"
+
+	ordererPubKey, ordererCSR, ordererSerial := createCSR(ctx, sess, orgName, "orderer")
+	adminPubKey, adminCSR, adminSerial := createCSR(ctx, sess, orgName, "admin")
+
+	// Retrieves org CA cert
+	orgCAPath := network.OrdererOrgCADir()
+	caBytes, err := ioutil.ReadFile(filepath.Join(orgCAPath, "ca.example.com-cert.pem"))
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Updating the orderer signcerts")
+
+	newOrdererPemCert := buildCert(network, caBytes, orgCAPath, ordererCSR, ordererSerial, ordererPubKey)
+
+	updateMSPFolder(network.OrdererLocalMSPDir(orderer), "orderer.example.com-cert.pem", newOrdererPemCert)
+
+	By("Updating the orderer admin user signcerts")
+	newAdminPemCert := buildCert(network, caBytes, orgCAPath, adminCSR, adminSerial, adminPubKey)
+
+	orgAdminMSPPath := network.OrdererUserMSPDir(orderer, "Admin")
+	updateMSPFolder(orgAdminMSPPath, "Admin@example.com-cert.pem", newAdminPemCert)
+
+}
+
+// Creates pkcs11 context and session
+func setupPKCS11Ctx(lib, label, pin string) (*pkcs11.Ctx, pkcs11.SessionHandle) {
+
+	ctx := pkcs11.New(lib)
+
+	ctx.Initialize()
+
+	slot := findPKCS11Slot(ctx, label)
+
+	sess, err := ctx.OpenSession(slot, pkcs11.CKF_SERIAL_SESSION|pkcs11.CKF_RW_SESSION)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Login
+	err = ctx.Login(sess, pkcs11.CKU_USER, pin)
+	Expect(err).NotTo(HaveOccurred())
+
+	return ctx, sess
+
+}
+
+// Identifies pkcs11 slot using specified label
+func findPKCS11Slot(ctx *pkcs11.Ctx, label string) uint {
+
+	var (
+		slot      uint
+		foundSlot bool
+	)
+
+	slots, err := ctx.GetSlotList(true)
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, s := range slots {
+		tokInfo, err := ctx.GetTokenInfo(s)
+		Expect(err).ToNot(HaveOccurred())
+
+		if tokInfo.Label == label {
+			foundSlot = true
+			slot = s
+			break
+		}
+	}
+
+	if !foundSlot {
+		Expect(foundSlot).To(BeTrue(), "Could not find slot with label %s", label)
+	}
+
+	return slot
+
+}
```

Comment:
I think this can be simplified a lot by returning immediately when the slot is found. After the for loop, we can have a simple `Fail(...)`.

Reply:
Oh irony. I asked you to assert on `foundSlot` but stepping back, you don't need that variable so...